### PR TITLE
Recover from a stale coalesced connection.

### PR DESCRIPTION
**Pull request by dave-r12**
https://github.com/square/okhttp/issues/3308

### CODE REVIEW

**Comment by dave-r12** [comment]
This seems like a corner case (maybe impossible?), and might be related to #3308.

File: okhttp-tests/src/test/java/okhttp3/ConnectionCoalescingTest.java [hunk 1]
```diff
@@ -117,6 +118,33 @@
...
+            connection.set(chain.connection());
+            return chain.proceed(chain.request());
+          }
+        })
+        .build();
+    dns.set("san.com", Dns.SYSTEM.lookup(server.getHostName()).subList(0, 1));
+
+    assert200Http2Response(execute(url), server.getHostName());
+
+    // Simulate a stale connection in the pool.
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/ConnectionCoalescingTest.java]: Neat

**Comment by swankjesse** [comment]
Nice find.
